kpi-tracker/src/main.rs

```diff
@@ -397,34 +405,106 @@ async fn process_genesis_block(
         .response;
 
     let block_details = BlockDetails {
-        block_time: block_info.block_slot_time,
-        height:     block_info.block_height,
+        block_time:  block_info.block_slot_time,
+        height:      block_info.block_height,
+        total_stake: None,
     };
 
     let genesis_accounts = accounts_in_block(node, block_hash).await?;
+
+    log::debug!("GENESIS BLOCK: {}\n{:?}", block_hash, block_details);
+    log::debug!(
+        "GENESIS ACCOUNTS:\n{}",
+        &genesis_accounts
+            .keys()
+            .map(|address| address.to_string())
+            .collect::<Vec<_>>()
+            .join("\n")
+    );
+
     init_db(db, (block_hash, block_details), genesis_accounts);
 
     Ok(())
 }
 
-/// Process a block, represented by `block_hash`, updating the `db`
-/// corresponding to events captured by the block.
-async fn process_block(
+async fn is_p4_payday_block(node: &mut Client, block_hash: BlockHash) -> anyhow::Result<bool> {
```

Comment:
You could use `get_tokenomics_info` to get the protocol version of the block.

Reply:
This also has the total stake, which is the whole reason for getting into this mess, and removes the need for this function. I'll refactor.